Applications/Spire/Source/Ui/QuantityInputWidget.cpp

```diff
@@ -1,46 +1,56 @@
 #include "Spire/Ui/QuantityInputWidget.hpp"
 #include <QHBoxLayout>
-#include <QRegularExpressionValidator>
+#include <QLineEdit>
 
 using namespace boost::signals2;
 using namespace Nexus;
 using namespace Spire;
 
-QuantityInputWidget::QuantityInputWidget(QWidget* parent)
-    : TextInputWidget(parent),
-      m_item_delegate(this) {
-  setValidator(new QRegularExpressionValidator(
-    QRegularExpression("^[0-9]*$"), this));
-  connect(this, &QLineEdit::editingFinished, this,
-    &QuantityInputWidget::on_line_edit_committed);
-  connect(this, &QLineEdit::textEdited, this,
-    &QuantityInputWidget::on_line_edit_modified);
-  m_locale.setNumberOptions(QLocale::OmitGroupSeparator);
+QuantityInputWidget::QuantityInputWidget(Nexus::Quantity value,
+    QWidget* parent)
+    : QWidget(parent) {
+  auto layout = new QHBoxLayout(this);
+  layout->setContentsMargins({});
+  m_input_widget = new DecimalInputWidget(0, this);
+  layout->addWidget(m_input_widget);
+  setFocusProxy(m_input_widget);
+  connect(m_input_widget, &DecimalInputWidget::editingFinished,
+    this, &QuantityInputWidget::on_editing_finished);
+  connect(m_input_widget, QOverload<double>::of(&QDoubleSpinBox::valueChanged),
+    [=] (auto value) { on_value_changed(); });
+}
+
+void QuantityInputWidget::resizeEvent(QResizeEvent* event) {
+  m_input_widget->setFixedSize(size());
+  QWidget::resizeEvent(event);
+}
+
+void QuantityInputWidget::set_minimum(Nexus::Quantity minimum) {
+  m_input_widget->setMinimum(static_cast<double>(minimum));
+}
+
+void QuantityInputWidget::set_maximum(Nexus::Quantity maximum) {
+  m_input_widget->setMaximum(static_cast<double>(maximum));
 }
 
 void QuantityInputWidget::set_value(Quantity value) {
-  setText(m_item_delegate.displayText(
-    QVariant::fromValue(Truncate(value, 0)), m_locale));
+  m_input_widget->setValue(static_cast<double>(value));
```

Comment:
This isn't possible, not all ```Quantity```'s can be expressed as a ```double```. This is a big issue because even a common value like ```1.1``` is a valid ```Quantity``` but not a valid ```double```.

Reply:
I borrowed this from the CustomVariantItemDelegate::displayText(...) function that converts a Quantity to a string. Would I have to go Quantity -> QString -> double to convert this properly?

Also, how is 1.1 not a valid double? Or is it just the context?

Reply:
There is no way to represent the value ```1.1``` as a double.

I can look into this a bit later today but I was hoping you could set the value of the ```QDoubleSpinBox``` using a ```QString``` instead of a ```double```.

Reply:
I can create a getter to access the underlying QLineEdit to set the current text, but there's still the issue of setting the range. There is a private implementation (QAbstractSpinBoxPrivate) that has a setRange() function that takes QVariants.

Reply:
Setting the range isn't a concern since there's no use case where a range is set to more than 15 decimal places.

Reply:
Sorry, the range for the spin box means the acceptable input range from the minimum to maximum value.

Would it be possible to just use the conversion functions given by the QDoubleSpinBox? Converting the Quantity to a QString then passing it to QDoubleSpinBox::valueFromText() would give the double that the input would return from the given text.

Reply:
Yep I got what you meant by range, it's not an issue for our use case.

Your solution of using ```valueFromText``` sounds good. Go with it.